src/install.ts

```diff
@@ -14,43 +12,35 @@ declare module 'vue/types/options' {
 
 export default function install(Vue: typeof VueClass, mobxMethods: IMobxMethods) {
 
-  const defineReactive = (Vue as any).util.defineReactive
+  const changeDetector = new ChangeDetector(Vue, mobxMethods)
 
   function beforeCreate(this: VueClass) {
     const vm = this
-    getFromStoreEntries(vm).forEach(({ key, compute }) => {
-      defineReactive(vm, key, null, null, true)
-    })
+
+    vm.$options.computed = getFromStoreEntries(vm).reduce(
+      (computed, { key, set }) => {
+        changeDetector.defineReactiveProperty(vm, key)
+
+        return Object.assign(
+          computed,
+          { [key]: createComputedProperty(changeDetector, vm, key, set) }
+        )
+      },
+      vm.$options.computed || {}
+    )
   }
 
   function created(this: VueClass) {
     const vm = this
-    const entries = getFromStoreEntries(vm)
-    if (entries.length <= 0) {
-      return
-    }
 
-    const disposers: Disposer[] = vm['__movueDisposers__'] = []
-
-    entries.forEach(({ key, compute }) => {
-      disposers.push(mobxMethods.reaction(
-        () => compute.apply(vm),
-        val => {
-          vm[key] = val
-        },
-        true
-      ))
-    })
+    changeDetector.defineReactionList(vm, getFromStoreEntries(vm))
   }
 
   function beforeDestroy(this: VueClass) {
     const vm = this
-    const disposers: Disposer[] = vm['__movueDisposers__']
-    if (disposers) {
-      disposers.forEach(
-        dispose => dispose()
-      )
-    }
+
+    changeDetector.removeReactionList(vm)
+    getFromStoreEntries(vm).map(({ key }) => changeDetector.removeReactiveProperty(vm, key))
```

Comment:
`forEach` will be better here than `map`

Reply:
I agree. Fixed that.